BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsDatabase.kt

```diff
@@ -5,13 +5,15 @@ import androidx.room.AutoMigration
 import androidx.room.Database
 import androidx.room.Room
 import androidx.room.RoomDatabase
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
 
 // Ref: https://developer.android.com/training/data-storage/room/migrating-db-versions
 @Database(
-        version = 2,
+        version = 3,
         entities = [AnalyticsEvent::class],
         autoMigrations = [
-            AutoMigration(from = 1, to = 2)
+            AutoMigration(from = 2, to = 3)
```

Comment:
AFAIK you _add_ to the auto migration, never replace. Are you sure you want to replace?
What if we have a user on version 1 and they skip several updates to update to this version. How would their db migrate?

Reply:
Thanks for the heads up! I've added a database migration to ensure functionality for merchants who might upgrade directly from version 1 to version 3. This approach is based on the Kotlin [documentation](https://developer.android.com/training/data-storage/room/migrating-db-versions) for database management. Since we are modifying the `AnalyticsEvent` entity, it's necessary to increase the database version number; otherwise, the SDK will throw an exception like this:

_java.lang.IllegalStateException: Room cannot verify the data integrity. Looks like you've changed schema but forgot to update the version number. You can simply fix this by increasing the version number._ 

With the version number incremented, we need to perform a database migration from v2 to v3 to avoid an exception like:

_java.lang.IllegalStateException: A migration from 2 to 3 was required but not found. Please provide the necessary Migration path via RoomDatabase.Builder.addMigration(Migration ...) or allow for destructive migrations via one of the RoomDatabase.Builder.fallbackToDestructiveMigration* methods._

Reply:
+1 to Sai's concern - I think what we want to do is is create a new auto migration from 2 to 3 (i.e. copy the AutoMigration from 1 to 2 and add another below it for the changes from 2 to 3)